Resources/public/inlineDeleteCheckboxes.js

```diff
@@ -0,0 +1,32 @@
+/*
+
+ This file is part of the Sonata package.
+
+ (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+ */
+
+var handleInlineDeleteCheckboxes = function() {
+    var checkboxes = jQuery('.sonata-admin-type-delete-checkbox');
+
+    if(checkboxes.length) {
+        var eventType = window.SONATA_CONFIG.USE_ICHECK ? 'ifChanged': 'change';
+
+        checkboxes.on(eventType, function() {
```

Comment:
If you bind an event to document with a selector you don't need to reinitialize the listener on dinamically added content.